src/internal/components/Draggable.tsx

```diff
@@ -0,0 +1,93 @@
+import { useCallback, useEffect, useState } from 'react';
+import { cn } from '../../styles/theme';
+
+type DraggableProps = {
+  children: React.ReactNode;
+  gridSize?: number;
+  startingPosition?: { x: number; y: number }; // TODO [BOE-886]: make this based on the parent component's position
+};
+
+export default function Draggable({
+  children,
+  gridSize = 1,
+  startingPosition = { x: 20, y: 20 },
+}: DraggableProps) {
+  const [position, setPosition] = useState(startingPosition);
+  const [dragOffset, setDragOffset] = useState({ x: 0, y: 0 });
+  const [isDragging, setIsDragging] = useState(false);
+
+  const snapToGrid = useCallback(
+    (positionValue: number) => {
+      return Math.round(positionValue / gridSize) * gridSize;
+    },
+    [gridSize],
+  );
```

Comment:
can this be disabled?   I believe touch or pointer events on mobile could return decimals.

Reply:
+ How does a developer disable this draggable functionality from a component?

Reply:
> How does a developer disable this draggable functionality from a component?

developer does not disable draggability on this component. this component is specifically for enabling draggability. if they don't want draggability, they should not use the component called `Draggable`.

> can this be disabled? I believe touch or pointer events on mobile could return decimals.

IIUC, you're referring to the snap-to-grid right? I didn't realize that decimals were possible so i didn't see a need to disable. i'll add that in

Reply:
on mobile where screens have a higher actual resolution then the pixel count they display this is possible.

Reply:
TIL, thank you! 

i've made the update so that snapToGrid is optional